sdk/textanalytics/ai-text-analytics/src/lro/analyze/operation.ts

```diff
@@ -0,0 +1,384 @@
+  TasksStateTasksKeyPhraseExtractionTasksItem,
+  TextDocumentBatchStatistics,
+  TextDocumentInput
+} from "../../generated/models";
+import {
+  AnalyzeResult,
+  AnalyzeResultsArray,
+  PagedAsyncIterableAnalyzeResults,
+  PaginatedAnalyzeResults
+} from "../../analyzeResult";
+import { PageSettings } from "@azure/core-paging";
+import {
+  addStrEncodingParam,
+  getJobID,
+  handleInvalidDocumentBatch,
+  nextLinkToTopAndSkip
+} from "../../util";
+import {
+  AnalysisPollOperation,
+  AnalysisPollOperationState,
+  PollingOptions,
+  TextAnalyticsStatusOperationOptions
+} from "../poller";
+import { GeneratedClient as Client } from "../../generated";
+import { CanonicalCode } from "@opentelemetry/api";
+import { createSpan } from "../../tracing";
+import { TextAnalyticsOperationOptions } from "../../textAnalyticsOperationOptions";
+import { makeRecognizeCategorizedEntitiesResult } from "../../recognizeCategorizedEntitiesResult";
+import { makeRecognizePiiEntitiesResult } from "../../recognizePiiEntitiesResult";
+import { makeExtractKeyPhrasesResult } from "../../extractKeyPhrasesResult";
+import { processAndcombineSuccessfulErroneousDocuments } from "../../textAnalyticsResult";
+export { State };
+
+interface AnalyzeResultsWithPagination {
+  result: AnalyzeResultsArray;
+  top?: number;
+  skip?: number;
+}
+
+interface AnalyzeJobStatus {
+  done: boolean;
+  /**
+   * Statistics about the input document batch and how it was processed
+   * by the service. This property will have a value when includeStatistics is set to true
+   * in the client call.
+   */
+  statistics?: TextDocumentBatchStatistics;
+}
+
+interface BeginAnalyzeInternalOptions extends OperationOptions {}
+
+/**
+ * Options for configuring analyze jobs.
+ */
+export interface AnalyzeJobOptions extends TextAnalyticsOperationOptions {}
+
+/**
+ * Options for the begin analyze operation.
+ */
+export interface BeginAnalyzeOptions {
+  polling?: PollingOptions;
+  analyze?: AnalyzeJobOptions;
+}
+
+/**
+ * The state of the begin analyze polling operation.
+ */
+export interface BeginAnalyzePollState
+  extends AnalysisPollOperationState<PaginatedAnalyzeResults> {}
+
+/**
+ * Class that represents a poller that waits for results of the analyze
+ * operation.
+ */
+export class BeginAnalyzePollerOperation extends AnalysisPollOperation<
+  BeginAnalyzePollState,
+  PaginatedAnalyzeResults
+> {
+  constructor(
+    public state: BeginAnalyzePollState,
+    private client: Client,
+    private documents: TextDocumentInput[],
+    private tasks: JobManifestTasks,
+    private options: BeginAnalyzeOptions = {},
+    private statusOptions: TextAnalyticsStatusOperationOptions
+  ) {
+    super(state);
+  }
+
+  /**
+   * should be called only after all the status of the analyze jobs became
+   * "succeeded" and it returns an iterator for the results and provides a
+   * byPage method to return the results paginated.
+   */
+  private listAnalyzeResultsByPage(
```

Comment:
I feel like this name is wrong, since it's not providing paged results, it's providing an iterator over **all** results. You have to call `.byPage` to actually get them in pages. Why not call it `listAnalyzeResults` ?

Reply:
Done.